pkg/runtime/config.go

```diff
@@ -77,46 +82,91 @@ type Config struct {
 	GracefulShutdownDuration     time.Duration
 	EnableAPILogging             bool
 	DisableBuiltinK8sSecretStore bool
+	AppHealthCheck               *apphealth.Config
+	AppHealthCheckHTTPPath       string
+}
+
+// NewRuntimeConfigOpts contains options for NewRuntimeConfig.
+type NewRuntimeConfigOpts struct {
+	ID                           string
+	PlacementAddresses           []string
+	controlPlaneAddress          string
+	AllowedOrigins               string
+	GlobalConfig                 string
+	ComponentsPath               string
+	AppProtocol                  string
+	Mode                         string
+	HTTPPort                     int
+	InternalGRPCPort             int
+	APIGRPCPort                  int
+	APIListenAddresses           []string
+	PublicPort                   *int
+	AppPort                      int
+	ProfilePort                  int
+	EnableProfiling              bool
+	MaxConcurrency               int
+	MTLSEnabled                  bool
+	SentryAddress                string
+	AppSSL                       bool
+	MaxRequestBodySize           int
+	UnixDomainSocket             string
+	ReadBufferSize               int
+	StreamRequestBody            bool
+	GracefulShutdownDuration     time.Duration
+	EnableAPILogging             bool
+	DisableBuiltinK8sSecretStore bool
+	EnableAppHealthCheck         bool
+	AppHealthCheckPath           string
+	AppHealthProbeInterval       time.Duration
+	AppHealthProbeTimeout        time.Duration
+	AppHealthThreshold           int32
 }
 
 // NewRuntimeConfig returns a new runtime config.
-func NewRuntimeConfig(
-	id string, placementAddresses []string, controlPlaneAddress, allowedOrigins, globalConfig, componentsPath,
-	appProtocol, mode string, httpPort, internalGRPCPort, apiGRPCPort int, apiListenAddresses []string, publicPort *int,
-	appPort, profilePort int, enableProfiling bool, maxConcurrency int, mtlsEnabled bool, sentryAddress string, appSSL bool,
-	maxRequestBodySize int, unixDomainSocket string, readBufferSize int, streamRequestBody bool, gracefulShutdownDuration time.Duration, enableAPILogging bool, disableBuiltinK8sSecretStore bool,
-) *Config {
+func NewRuntimeConfig(opts NewRuntimeConfigOpts) *Config {
```

Comment:
Why this change: the function already had almost 30 parameters, which were hard to maintain ("what is this 'false' for?". And I had to add 5 more...